scripts/dist/debian-install.sh

```diff
@@ -0,0 +1,148 @@
+
+#!/bin/bash
+
+# Exit on any errors
+set -e
+
+# Color codes for output messages
+RED="\033[0;31m"
+GREEN="\033[0;32m"
+YELLOW="\033[1;33m"
+NC="\033[0m" # No Color
+
+# Default values for flags
+subscriptionID=""
+locationName=""
+subscriptionAPIKey=""
+instDir="/opt/shield/scripts"
+downloadURL="https://raw.githubusercontent.com/Shield-Cyber/ShieldCyber-REST-API/main/scripts/"
+
+# Help message for usage
+usage() {
+    echo -e "${YELLOW}Usage: $0 [options]${NC}"
+    echo "Options:"
+    echo "  -s    Subscription ID"
+    echo "  -l    Location Name"
+    echo "  -k    Subscription API Key"
+    echo "  -h    Display this help message"
+    exit 1
+}
+
+# Parse command line options
+while getopts ":s:l:k:h" opt; do
+    case ${opt} in
+        s )
+            subscriptionID=$OPTARG
+            ;;
+        l )
+            locationName=$OPTARG
+            ;;
+        k )
+            subscriptionAPIKey=$OPTARG
+            ;;
+        h )
+            usage
+            ;;
+        \? )
+            echo -e "${RED}Invalid Option: -$OPTARG${NC}" 1>&2
+            usage
+            ;;
+        : )
+            echo -e "${RED}Invalid Option: -$OPTARG requires an argument${NC}" 1>&2
+            usage
+            ;;
+    esac
+done
+shift $((OPTIND -1))
+
+# Function to ensure directory exists
+ensure_directory() {
+    local directory=$1
+
+    if [ ! -d "$directory" ]; then
+        mkdir -p "$directory"
+        echo -e "${YELLOW}$directory created.${NC}"
+    fi
+}
+
+# Function to download a file
+download_file() {
+    local url=$downloadURL$1
+    local filename=$(basename "$url")
+    local filepath="$instDir/$filename"
+
+    ensure_directory "$instDir"
+
+    echo -e "${YELLOW}Downloading $filename to $instDir...${NC}"
+    curl -sS "$url" -o "$filepath"
+    echo -e "${GREEN}$filename downloaded to $instDir!${NC}"
+}
+
+# Install prerequisites
+sudo apt update -y && sudo apt upgrade -y 
+echo "Installing prerequisites..."
+sudo apt update -y
+sudo apt install wget -y
+curl -o packages-microsoft-prod.deb https://packages.microsoft.com/config/debian/12/packages-microsoft-prod.deb # Someone needs to make this dynamic and check the OS version.
```

Comment:
@codecadwallader this *may* need an update to dynamically check for the correct version of Debian. Currently its set to use 12. Since Debian does not come with the Microsoft repos they need to be added manually.